### PR TITLE
Fixing errors in VIP environment as well as hooking the plugin settin…

### PR DESCRIPTION
**Pull request by bcampeau**
…gs page appropriately

### CODE REVIEW

File: php/class-wp-seo-settings.php [hunk 1]
```diff
@@ -202,7 +202,7 @@ public function get_archived_post_types() {
 	 * @return array With slugs of any enabled taxonomies.
 	 */
 	public function get_enabled_taxonomies() {
-		return $this->get_option( 'taxonomies' );
+		return $this->get_option( 'taxonomies', array() );
```
Comment (dlh01) [inline php/class-wp-seo-settings.php]: This is calling the class `get_option()`, not core; the class doesn't currently support a default.

Comment (bcampeau) [inline php/class-wp-seo-settings.php]: Oops, didn't throw an error though!

File: php/class-wp-seo.php [hunk 2]
```diff
@@ -251,12 +251,15 @@ public function save_post_fields( $post_id ) {
...
-			add_action( $slug . '_add_form_fields', array( $this, 'add_term_meta_fields' ), 10, 2 );
-			add_action( $slug . '_edit_form', array( $this, 'edit_term_meta_fields' ), 10, 2 );
+		$taxonomies = WP_SEO_Settings()->get_enabled_taxonomies();
+		if ( ! empty( $taxonomies ) ) {
+			foreach ( $taxonomies as $slug ) {
+				add_action( $slug . '_add_form_fields', array( $this, 'add_term_meta_fields' ), 10, 2 );
+				add_action( $slug . '_edit_form', array( $this, 'edit_term_meta_fields' ), 10, 2 );
+			}
+			add_action( 'created_term', array( $this, 'save_term_fields' ), 10, 3 );
+			add_action( 'edited_term', array( $this, 'save_term_fields' ), 10, 3 );
```
Comment (dlh01) [inline php/class-wp-seo.php]: You might need to make a similar change to `WP_SEO_Settings::has_post_fields()` and `WP_SEO_Settings::has_term_fields()`, both of which immediately call `in_array()`. (Alternatively, add a `! empty()` check to `get_enabled_taxonomies()` and `get_enabled_post_types()`.)

File: php/class-wp-seo-settings.php [hunk 3]
```diff
@@ -794,4 +794,4 @@ public function view_formatting_tags_help_tab() {
 function WP_SEO_Settings() {
 	return WP_SEO_Settings::instance();
 }
-add_action( 'plugins_loaded', 'WP_SEO_Settings' );
\ No newline at end of file
+add_action( 'after_setup_theme', 'WP_SEO_Settings' );
```
Comment (dlh01) [inline php/class-wp-seo-settings.php]: If memory serves, this needs to be at priority 9, because `WP_SEO` (at priority 10) depends on it.

**Comment by dlh01** [comment]
One note about hook priorities for your consideration, otherwise :deciduous_tree:

**Comment by bcampeau** [comment]
Changed the priority, just waiting on Travis and then I'll resubmit to VIP.

**Comment by dlh01** [comment]
Cool, thanks!

**Pull request merged**